plugins/palo_alto_cortex_xdr/icon_palo_alto_cortex_xdr/tasks/monitor_alerts/task.py

```diff
@@ -0,0 +1,346 @@
+            log_msg += "No previous alert time within state. "
+            custom_timings = custom_config.get(LAST_ALERT_TIME, {})
+            custom_date = custom_timings.get("date")
+            custom_hours = custom_timings.get("hours", DEFAULT_LOOKBACK_HOURS)
+            start = datetime(**custom_date) if custom_date else (dt_now - timedelta(hours=custom_hours))
+            state[LAST_ALERT_TIME] = start.strftime(TIME_FORMAT)
+        else:
+            # check if we have held the TS beyond our max lookback
+            lookback_days = custom_config.get(f"{LAST_ALERT_TIME}_days", MAX_LOOKBACK_DAYS)
+            default_date_lookback = dt_now - timedelta(days=lookback_days)  # if not passed from CPS create on the fly
+            custom_lookback = custom_config.get(f"max_{LAST_ALERT_TIME}", {})
+            comparison_date = datetime(**custom_lookback) if custom_lookback else default_date_lookback
+            comparison_date = comparison_date.replace(tzinfo=timezone.utc).strftime(TIME_FORMAT)
+            if comparison_date > saved_time:
+                self.logger.info(f"Saved time ({saved_time}) exceeds cut off, moving to ({comparison_date}).")
+                state[LAST_ALERT_TIME] = comparison_date
+                # pop state held time filters if they exist, incase customer has paused integration when paginating
+                # state = self._drop_pagination_state(state)
+
+        start_time = state.get(LAST_ALERT_TIME)
+        self.logger.info(f"{log_msg}Applying the following start time='{start_time}'. Limit={alert_limit}.")
+
+        start_time = int(datetime.strptime(start_time, "%Y-%m-%dT%H:%M:%S.%fZ").timestamp()) * 1000
+
+        return start_time, alert_limit
+
+    ###########################
+    # Handle Pagination
+    ###########################
+    # def _drop_pagination_state(self, state: Dict[str, str]) -> Dict[str, str]:
+    #     """
+    #     Helper function to pop values from the state if we need to break out of pagination.
+    #     :return: state
+    #     """
+    #     log_msg = "Dropped the following keys from state: "
+    #     if state.get(TO_TIME_FILTER):
+    #         log_msg += f"{TO_TIME_FILTER}; "
+    #         state.pop(TO_TIME_FILTER)
+    #
+    #     if state.get(FROM_TIME_FILTER):
+    #         log_msg += f"{FROM_TIME_FILTER}; "
+    #         state.pop(FROM_TIME_FILTER)
+    #
+    #     self.logger.debug(log_msg)
+    #
+    #     return state
+
+    ###########################
+    # Make request
+    ###########################
+    def get_response(self, post_body: dict):
+        """
+        Helper method to return the response JSON and total count
+        """
+        headers = self.connection.xdr_api.get_headers()
+
+        fqdn = self.connection.xdr_api.get_url()
+        url = f"{fqdn}{self.endpoint}"
+
+        request = requests.Request(method="post", url=url, headers=headers, json=post_body)
+        response = make_request(request, timeout=120)
```

Comment:
You'll need to set the data_location here so that the response can be accessed in the exception 

i.e. 

```
        response = make_request(
            timeout=60,
            exception_data_location=ResponseExceptionData.RESPONSE,
        )
```